teuthology/task/install.py

```diff
@@ -114,75 +114,78 @@ def _get_uri(tag, branch, sha1):
     :param remote: the teuthology.orchestra.remote.Remote object
     :param config: the config dict
     :returns: dict -- the information you want.
     """
-    retval = {}
-    relval = None
-    r = remote.run(
-        args=['arch'],
-        stdout=StringIO(),
-    )
-    retval['arch'] = r.stdout.getvalue().strip()
-    r = remote.run(
-        args=['lsb_release', '-is'],
-        stdout=StringIO(),
-    )
-    retval['distro'] = r.stdout.getvalue().strip()
-    r = remote.run(
-        args=[
-            'lsb_release', '-rs'], stdout=StringIO())
-    retval['relval'] = r.stdout.getvalue().strip()
-    dist_name = None
-    if retval['distro'] in ('CentOS', 'RedHatEnterpriseServer'):
-        relval = retval['relval']
-        relval = relval[0:relval.find('.')]
-        distri = 'centos'
-        retval['distro_release'] = '%s%s' % (distri, relval)
-        retval['dist'] = retval['distro_release']
-        dist_name = 'el'
-        retval['dist_release'] = '%s%s' % (dist_name, relval)
-    elif retval['distro'] == 'Fedora':
-        distri = retval['distro']
-        dist_name = 'fc'
-        retval['distro_release'] = '%s%s' % (dist_name, retval['relval'])
-        retval['dist'] = retval['dist_release'] = retval['distro_release']
+    result = dict()
+    result['arch'] = remote.arch
+
+    distro = remote.os.name
+    version = _get_gitbuilder_version(remote.os.version)
+    if distro in ('centos', 'rhel'):
+        distro = "centos"
+        dist_release = "el{0}".format(version)
+    elif distro == "fedora":
+        distro = "fc"
+        dist_release = "fc{0}".format(version)
     else:
-        r = remote.run(
-            args=['lsb_release', '-sc'],
-            stdout=StringIO(),
-        )
-        retval['dist'] = r.stdout.getvalue().strip()
-        retval['distro_release'] = None
-        retval['dist_release'] = None
+        distro = remote.os.codename
+        dist_release = distro
+        version = ""
+
+    result['dist'] = "{distro}{version}".format(
+        distro=distro,
+        version=version,
+    )
+    # this is used when constructing the rpm name
+    result["dist_release"] = dist_release
 
     # branch/tag/sha1 flavor
-    retval['flavor'] = config.get('flavor', 'basic')
+    result['flavor'] = config.get('flavor', 'basic')
 
     log.info('config is %s', config)
     tag = _get_config_value_for_remote(ctx, remote, config, 'tag')
     branch = _get_config_value_for_remote(ctx, remote, config, 'branch')
     sha1 = _get_config_value_for_remote(ctx, remote, config, 'sha1')
     uri = _get_uri(tag, branch, sha1)
-    retval['uri'] = uri
+    result['uri'] = uri
+
+    return result
+
+
+def _get_gitbuilder_version(version):
+    """
+    Parses a distro version string and returns a modified string
+    that matches the format needed for the gitbuilder url.
+
+    Minor version numbers are ignored if they end in a zero. If they do
+    not end in a zero the minor version number is included with a dash as
```

Comment:
Ah, yeah.  Originally I had used a dash, but during testing I figured out
that was wrong.  These docs just never got updated.  I'll do that now.

On Mon, Aug 3, 2015 at 2:57 PM, Zack Cerza <notifications@github.com> wrote:

> In teuthology/task/install.py
> <https://github.com/ceph/teuthology/pull/580#discussion_r36124126>:
>
> >
> >      log.info('config is %s', config)
> >      tag = _get_config_value_for_remote(ctx, remote, config, 'tag')
> >      branch = _get_config_value_for_remote(ctx, remote, config, 'branch')
> >      sha1 = _get_config_value_for_remote(ctx, remote, config, 'sha1')
> >      uri = _get_uri(tag, branch, sha1)
> > -    retval['uri'] = uri
> > +    result['uri'] = uri
> > +
> > +    return result
> > +
> > +
> > +def _get_gitbuilder_version(version):
> > +    """
> > +    Parses a distro version string and returns a modified string
> > +    that matches the format needed for the gitbuilder url.
> > +
> > +    Minor version numbers are ignored if they end in a zero. If they do
> > +    not end in a zero the minor version number is included with a dash as
>
> You say 'dash' but then you use an underscore!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ceph/teuthology/pull/580/files#r36124126>.
>